jade/wsgi/routes/v1/judgements.py

```diff
@@ -2,49 +2,78 @@
 
 import flask
 
-from func_tools import wraps
+from ... import responses, util
+from .... import types
 
-from ... import responses
 
+def configure(bp, config, trusted_clients, centralauth, state, stream):
 
-def configure(bp, trusted_clients, source):
+    @bp.route("/v1/judgements/<string:context>", methods=["get"])
+    def get_context_judgements(context):
+        """Gets all judgements within a context"""
+        judgements_doc = state.judgements.get(context)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<string:context>/<string:type>", methods=["get"])
+    def get_entity_type_judgements(context, type_):
+        """Gets all judgements for a specific entity type"""
+        judgements_doc = state.judgements.get(context, type_)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<string:context>/<string:type>/<int:id>", methods=["get"])  # noqa
+    def get_entity_judgements(context, type_, id_):
+        """Gets all judgements for a specific entity"""
+        judgements_doc = state.judgements.get(
+            context, type_, id_)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<string:context>/<string:type>/<int:id>/<string:schema>", methods=["get"])  # noqa
+    def get_entity_schema_judgements(context, type_, id_, schema):
+        """Gets all judgements for a specific entity schema"""
+        judgements_doc = state.judgements.get(
+            context, type_, id_, schema)
+        return flask.json.jsonify(judgements_doc)
+
+    @bp.route("/v1/judgements/<int:judgement_id>/preference", methods=["put"])
+    @util.authorized_user_action(
+        config, trusted_clients, centralauth, "set_judgement_preference")
+    def set_judgement_preference(judgement_id, gu_id, request_values):
+        """
+        Sets the preference bit for a specific judgement and returns the event
+        """
+        preference = json.loads(request_values['preference'])
+        if preference not in (True, False):
+            return responses.malformed_request(
+                "'preference' must be either 'true' or 'false'")
+
+        # Construct a new proto-event for the judgement
+        proto_event = types.events.JudgementPreferenceSet.proto(
+            gu_id=gu_id, judgement_id=judgement_id, preference=preference)
+
+        # Execute the proto_event and construct full event.
+        return util.execute_and_log_or_error(state, proto_event)
+
+    @bp.route("/v1/judgements/", methods=["post"])
+    @util.authorized_user_action(
```

Comment:
Per our IRC conversation, maybe we need to check authorization based on the wiki context instead.

Reply:
Right.  We should be able to access the request values (that include "context") inside of the authorized_user_action decorator.

Reply:
That would work, and I guess it's fine to split request parsing, it's a good trade-off in exchange for nice encapsulation of the security aspect.

Reply:
Note that authorized_user_action does all of the request pre-processing too (since that's necessary for decrypting data from trusted clients).  So in a way, we're just sequencing request parsing.  The request_values map roughly mimics flask.request.values with the same accessor methods (`MultiDict`).